packages/extension-utils/src/OAuthConfigForm/OAuthConfigForm.spec.tsx

```diff
@@ -0,0 +1,241 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2023 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+import React from 'react'
+import { fireEvent, screen, waitFor } from '@testing-library/react'
+import { renderWithTheme } from '@looker/components-test-utils'
+import userEvent from '@testing-library/user-event'
+import type { IApiSettings } from '@looker/sdk-rtl'
+import {
+  DefaultSettings,
+  BrowserTransport,
+  BrowserSession,
+} from '@looker/sdk-rtl'
+import { functionalSdk40 } from '@looker/sdk'
+import { OAuthConfigForm } from '.'
+import {
+  BrowserAdaptor,
+  registerTestEnvAdaptor,
+  OAuthConfigProvider,
+} from '@looker/extension-utils'
+
+jest.mock('react-router-dom', () => {
+  const ReactRouterDOM = jest.requireActual('react-router-dom')
+  return {
+    ...ReactRouterDOM,
+    useLocation: () => ({
+      pathname: '/',
+    }),
+    useHistory: jest
+      .fn()
+      .mockReturnValue({ push: jest.fn(), location: globalThis.location }),
+  }
+})
+
+const ConfigKey = 'ConfigKey'
+
+const initSdk = () => {
```

Comment:
You probably already have a test utils section planned where this could be shared, but we should in the future

Reply:
yeah. Also `RunIt` also has its own `initRunItSdk`. No reason why we shouldn't use this for all dual mode extensions. It will require a bit more configurability.